dll/win32/shell32/compat.h

```diff
@@ -0,0 +1,125 @@
+/*
+ * PROJECT:     ReactOS Shell
+ * LICENSE:     LGPL-2.0-or-later (https://spdx.org/licenses/LGPL-2.0-or-later)
+ * PURPOSE:     Absorbing version differences for compatibility
+ * COPYRIGHT:   Copyright 2023 Katayama Hirofumi MZ <katayama.hirofumi.mz@gmail.com>
+ */
+
+#pragma once
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#if (_WIN32_WINNT < _WIN32_WINNT_VISTA)
+    /*
+     * For versions < Vista+, redefine ShellMessageBoxW to ShellMessageBoxWrapW
+     * (this is needed to avoid a linker error). On Vista+ onwards, shell32.ShellMessageBoxW
+     * redirects to shlwapi.ShellMessageBoxW so the #define should not be needed.
+     */
+    #define ShellMessageBoxW ShellMessageBoxWrapW
+
+    LSTATUS
+    WINAPI
+    RegGetValueA(
```

Comment:
I suggest enabling all of them in our psdk. There is no point in duplicating them, just to be "compatible" with XP SDK.